python/tk_shell/task.py

```diff
@@ -1,55 +1,53 @@
 # Copyright (c) 2013 Shotgun Software Inc.
-# 
+#
 # CONFIDENTIAL AND PROPRIETARY
-# 
-# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
 # Source Code License included in this distribution package. See LICENSE.
-# By accessing, using, copying or modifying this work you indicate your 
-# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import tank
-import os
-import sys
-import threading
 
-from tank.platform.qt import QtCore, QtGui
+from tank.platform.qt import QtCore
+
 
 class Task(QtCore.QObject):
     """
     This is a wrapper class which allows us to run tank commands
     inside the QT universe. This approach is handy when an engine needs
     to start up a qt event loop as part of its initailization.
     """
+
     finished = QtCore.Signal()
 
     def __init__(self, engine, callback, args):
-        QtCore.QObject.__init__(self)        
+        QtCore.QObject.__init__(self)
         self._callback = callback
         self._args = args
         self._engine = engine
-        
+
     def run_command(self):
-        
+
         try:
             # execute the callback
             self._callback(*self._args)
-            
-        except tank.TankError, e:
+
+        except tank.TankError as e:
```

Comment:
Here is the culprit! The only line that wasn't Python 3 compatible. 😄

Reply:
😆Nice!